Source/JavaScriptCore/runtime/JSGenericTypedArrayViewPrototypeFunctions.h

```diff
@@ -250,12 +250,24 @@ ALWAYS_INLINE EncodedJSValue genericTypedArrayViewProtoFuncIncludes(VM& vm, JSGl
         }
     }
 
-    if constexpr (ViewClass::elementSize == 1 && ViewClass::Adaptor::isInteger) {
-        if (index < length) {
-            auto* result = static_cast<typename ViewClass::ElementType*>(memchr(array + index, targetOption.value(), length - index));
-            return JSValue::encode(jsBoolean(!!result));
+    if constexpr (ViewClass::Adaptor::isInteger) {
+        if constexpr (ViewClass::elementSize == 1) {
+            if (index < length) {
+                auto* result = static_cast<typename ViewClass::ElementType*>(memchr(array + index, targetOption.value(), length - index));
+                return JSValue::encode(jsBoolean(!!result));
+            }
+            return JSValue::encode(jsBoolean(false));
         }
-        return JSValue::encode(jsBoolean(false));
+
+#if CPU(ARM64)
+        if constexpr (ViewClass::elementSize == 2) {
+            if (index < length) {
+                auto* result = bitwise_cast<typename ViewClass::ElementType*>(WTF::memchr16(bitwise_cast<const uint16_t*>(array + index), targetOption.value(), length - index));
```

Comment:
This code is so similar to the above, wish we could share more instead of typing it all out twice.

Reply:
I'll attempt to unify these things once we implement all necessary fast pathes (memchr32, memchr64).

Reply:
Another nice thing to consider is to make these more generally usable by implementing them without SIMD too. After all, we have the 8-bit one as the memchr in the standard library in both non-SIMD and SIMD versions, although without any inlining it’s not so efficient for small lengths. I suspect there are places outside the JavaScript engine that could benefit from these.

Reply:
That's interesting idea! Probably, we can remove `HAVE(MEMCHR16)` and always use `memchr16`, which can be implemented in SIMD, or implemented in simple loop.

Reply:
Writing SIMD code is one of the best hobby for weekends!

Reply:
Just realized that the code I am working on in my other patch currently uses a memchr16 to find the null character at the end of a string.

Reply:
BTW, I wonder if we can use wmemchr for memchr32 since sizeof(wchar_t) is 4 on darwin. But it was super naive implementation, so if we want, we need to have our own :)